client/src/App.tsx

```diff
@@ -46,7 +46,27 @@ const ProtectedRoute: React.FC<ProtectedRouteProps> =  ({ children }) => {
 };
 
 function App() {
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    // Simulating an async operation (e.g., fetching data)
+    const fetchData = async () => {
+      // Assume the loading time is 2 seconds
+      setTimeout(() => {
+        setLoading(false);
+      }, 2000);
+    };
+
+    fetchData();
+  }, []);
+
   return (
+    <div>
+    {loading ? (
+      <div className="spinner-container">
+      <div className="spinner"></div>
```

Comment:
This looks great! Let's move the css and tsx code into a Spinner Component, in the components folder.